astropy/wcs/wcs.py

```diff
@@ -2446,6 +2450,26 @@ def sip_foc2pix(self, *args):
         """.format(docstrings.TWO_OR_MORE_ARGS('2', 8),
                    docstrings.RETURNS('pixel coordinates', 8))
 
+    def proj_plane_pixel_scales(self):
+        """Same as :func:`astropy.wcs.utils.proj_plane_pixel_scales`
+        but returns a `~astropy.units.Quantity`.
+
+        """
+        from astropy.wcs import utils as wcs_utils  # Avoid circular import
+        val = wcs_utils.proj_plane_pixel_scales(self)
+        units = [u.Unit(x) for x in self.wcs.cunit]
+        return u.Quantity(list(map(operator.mul, val, units)))
+
+    def proj_plane_pixel_area(self):
+        """Same as :func:`astropy.wcs.utils.proj_plane_pixel_area`
+        but returns a `~astropy.units.Quantity`.
+
+        """
+        from astropy.wcs import utils as wcs_utils  # Avoid circular import
+        val = wcs_utils.proj_plane_pixel_area(self)
+        unit = u.Unit(self.wcs.cunit[0]) * u.Unit(self.wcs.cunit[1])  # 2D only
```

Comment:
This should probably error if not 2D?

Reply:
`astropy.wcs.utils.proj_plane_pixel_area` already checks it.

Reply:
Is this check insufficient?

https://github.com/astropy/astropy/blob/4eef52a1c95471b837949c5a8f862a01e4ca1bfa/astropy/wcs/utils.py#L390-L392

If we need to add more check, I feel like it has to be added inside the `utils` function, not the high-level method wrapper.